### PR TITLE
DO NOT MERGE 324 replace prism js with shiki

### PR DESCRIPTION
**Pull request by DavieReid**
Hi @origami-z

### CODE REVIEW

**Comment by DavieReid** [comment]
> Depending on how long it takes to get Shiki to work, it would be great to have the commits where you added the languages to the docs in a separate PR. The code changes will still be highlighted by Prism so the site will look better.

I don't think the way languages are specified is the same...is it?

File: packages/site/next.config.js [hunk 1]
```diff
@@ -36,7 +38,15 @@ module.exports = {
     // Swaps out Buble for a smaller version that removes the latest Regex spec features.
     // See https://github.com/FormidableLabs/react-live#what-bundle-size-can-i-expect
     config.plugins.push(
-      new webpack.NormalModuleReplacementPlugin(/^buble$/, require.resolve('@philpl/buble'))
+      new webpack.NormalModuleReplacementPlugin(/^buble$/, require.resolve('@philpl/buble')),
+      new CopyPlugin({
+        patterns: [
+          {
+            from: path.posix.resolve(path.posix.dirname(require.resolve('shiki')), '..'),
```
Comment (mark-tate) [inline packages/site/next.config.js:45]: potentially another windows issue

I'm assuming we are going to go down the container route for local development.

**Review by mark-tate — APPROVED** [review]

**Comment by lilyvc** [comment]
> > Depending on how long it takes to get Shiki to work, it would be great to have the commits where you added the languages to the docs in a separate PR. The code changes will still be highlighted by Prism so the site will look better.
> 
> I don't think the way languages are specified is the same...is it?

oh you're right there is some extra syntax that would break our pages..
some of it would work like ```js but yea not worth separating

File: packages/site-middleware/src/plugins/prettyCode.ts [hunk 2]
```diff
@@ -1,4 +1,11 @@
+import { getHighlighter } from 'shiki';
+
 const options = {
+  getHighlighter: highlighterOptions =>
+    getHighlighter({
+      ...highlighterOptions,
+      langs: ['bat', 'diff', 'css', 'json', 'jsx', 'tsx', 'mdx', 'yaml', 'ts']
```
Comment (DavieReid) [inline packages/site-middleware/src/plugins/prettyCode.ts:7]: I don't think we use any other languages in code samples on the site...

@lilyvc

Comment (lilyvc) [reply packages/site-middleware/src/plugins/prettyCode.ts:7]: maybe js? for a config.js

File: packages/site-middleware/src/plugins/prettyCode.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+const options = {
+  getHighlighter: highlighterOptions =>
+    getHighlighter({
+      ...highlighterOptions,
+      langs: ['bat', 'diff', 'css', 'json', 'jsx', 'tsx', 'mdx', 'yaml', 'ts']
+    }),
+  theme: {
+    dark: 'dark-plus',
+    light: 'light-plus'
```
Comment (lilyvc) [inline packages/site-middleware/src/plugins/prettyCode.ts:11]: I don't love how the light theme looks, is it easy to switch it out?

File: packages/site-middleware/src/plugins/prettyCode.ts (hunk 2 above)
Comment (DavieReid) [reply packages/site-middleware/src/plugins/prettyCode.ts:7]: Good shout

File: packages/site-middleware/src/plugins/prettyCode.ts (hunk 3 above)
Comment (DavieReid) [reply packages/site-middleware/src/plugins/prettyCode.ts:11]: Very easy to change. I'm not a huge fan either but picked this with a11y in mind.

**Pull request closed without merging**